mondaycom/monday-ui-react-core
# fix(Button): differentiate different states of the button with a key

This [issue](https://github.com/facebook/react/issues/11538) was a very big pointer and helped reproduce this issue because it gives the same error message (and might be also the same issue - need further investigation to verify).

How to reproduce:
1. Install Google Translate plugin in [Chrome](https://chrome.google.com/webstore/detail/google-translate/aapbdbdomjkkjkaonfhkkikfgjllcleb/related).
2. Translate a page with a button (that changes to a loading state on click) to any language.
3. Click on the button, notice the page crushes.

Conjecture: 
Having three different root nodes that are returned from the Button confuses React understanding the changes in the component since all these root nodes get the same props.

The simplest solution (suggested by @orrgottlieb 👑 ) is to add a `key` for the different states.
Moving forward we might want to merge the different root nodes into a single one to make the component more coherent.
Also, since it caused the button to re-render and lose the CSS variable with the saved button width, I used a text placeholder instead, so the resize observer is redundant now. This is the same technique that is used in the loading state.

https://monday.monday.com/boards/3532714909/pulses/5303882985
